### ISSUE TITLE
Support environment variables in DockerComposeEnvironment

### ISSUE THREAD

**Issue by jmacmahon** [testcontainers/testcontainers-node#185]
You can do this for `GenericContainer` by using `.withEnv`, but there is nothing similar for a `DockerComposeEnvironment`

**Comment by cristianrgreco** [comment]
Hi @jmacmahon, any reason not to have the env variables defined in the `docker-compose.yml` file directly?

**Comment by jmacmahon** [comment]
We would like to specify a temporary source directory for a bind mount defined in the `docker-compose.yml` file.

A snippet from the file therefore looks like:
```
    volumes:
      - type: bind
        source: ${BIND_SOURCE}
        target: /whatever
```

Our test code will create the temporary directory somewhere on the host, and pass it down to `docker-compose` by setting the appropriate environment variable.

For now, we have worked around this by setting `process.env.BIND_SOURCE = cratedTemporaryDirectory` before calling `new DockerComposeEnvironment()`, but this is not ideal.

From this issue: https://github.com/PDMLab/docker-compose/issues/7, it looks like one can simply pass the `env` parameter to the `upAll` function in the `docker-compose` npm package, which I see is what this project uses.

**Comment by cristianrgreco** [comment]
@jmacmahon Thanks for explaining.

Would something like this work? I've deliberately kept the API the same to that of the `GenericContainer`:

```typescript
const environment = await new DockerComposeEnvironment(composeFilePath, composeFile)
  .withEnv("ENV_KEY_1", "ENV_VALUE")
  .withEnv("ENV_KEY_2", "ENV_VALUE")
  .up();
```

**Comment by jmacmahon** [comment]
> ```ts
> const environment = await new DockerComposeEnvironment(composeFilePath, composeFile)
>   .withEnv("ENV_KEY_1", "ENV_VALUE")
>   .withEnv("ENV_KEY_2", "ENV_VALUE")
>   .up();
> ```

That looks great 👍

**Comment by cristianrgreco** [comment]
@jmacmahon Just to double check, here's an excerpt from the README in the PR, does it look good to you?

> Bind environment variables to the docker-compose file. For example if we have the following docker-compose file:
> 
> ```yaml
> services:
>   redis:
>     image: redis:${TAG}
> ```
> 
> Then we can set `TAG` as follows:
> 
> ```javascript
> const { DockerComposeEnvironment } = require("testcontainers");
>
> const environment = await new DockerComposeEnvironment(composeFilePath, composeFile)
>   .withEnv("TAG", "value")
>   .up();
> ```

If so I'll merge and release

**Comment by jmacmahon** [comment]
> @jmacmahon Just to double check, here's an excerpt from the README in the PR, does it look good to you?

Looks good to me, thanks for doing it so quick! Great package 😊

**Comment by cristianrgreco** [comment]
Great, thanks! Released in [v7.3.0](https://github.com/testcontainers/testcontainers-node/releases/tag/v7.3.0) 🙂
